### PR TITLE
feat: new gofmt lint

### PR DESCRIPTION
**Pull request by wwcchh0123**

### CODE REVIEW

File: internal/linters/go/gofmt/test.txt
Comment (CarlJi) [inline internal/linters/go/gofmt/test.txt:1]: test data can be put in `testdata` folder, refer: https://pkg.go.dev/cmd/go/internal/test#section-documentation

Comment (wwcchh0123) [reply internal/linters/go/gofmt/test.txt:1]: updated

**Comment by wwcchh0123** [comment]
The effects of the gofmt can be found in the following PR: https://github.com/wwcchh0123/testrepo/pull/2

**Comment by wwcchh0123** [comment]
The  effects of the github suggestions comment can be found in the following PR:https://github.com/wwcchh0123/testrepo/pull/4

File: internal/linters/go/gofmt/gofmt_test.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/qiniu/x/xlog"
+)
+
+func TestFormatGofmt(t *testing.T) {
```
Comment (CarlJi) [inline internal/linters/go/gofmt/gofmt_test.go:11]: This case need assert function

File: internal/linters/go/gofmt/gofmt.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+				}
+
+				//Parameter Reset
+				message = " Is your code not properly formatted? Here are some suggestions below\n```suggestion"
+				locationLine = 0
+				lineNumber = 0
+				nonFirstTime = true
+
+				//Extract row information
+				re := regexp.MustCompile(`-(\d+),(\d+)`)
```
Comment (CarlJi) [inline internal/linters/go/gofmt/gofmt.go:166]: Compile the expression once

Comment (wwcchh0123) [reply internal/linters/go/gofmt/gofmt.go]: Each diff massage should compile onece

File: internal/linters/go/gofmt/gofmt_test.go (hunk 1 above)
Comment (wwcchh0123) [reply internal/linters/go/gofmt/gofmt_test.go]: updated

File: internal/linters/go/gofmt/gofmt_test.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/qiniu/reviewbot/internal/linters"
+	"github.com/qiniu/x/xlog"
+)
+
+func TestGofmtOutput(t *testing.T) {
+	content, err := os.ReadFile("../../../../testdata/gofmt_test.txt")
```
Comment (CarlJi) [inline internal/linters/go/gofmt/gofmt_test.go:13]: Put testdata folder under this package

File: internal/linters/go/gofmt/gofmt_test.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	if err != nil {
+		fmt.Println("无法读取文件:", err)
```
Comment (CarlJi) [inline internal/linters/go/gofmt/gofmt_test.go:15]: assert error

Comment (wwcchh0123) [reply internal/linters/go/gofmt/gofmt_test.go]: updated

File: internal/linters/go/gofmt/gofmt_test.go (hunk 3 above)
Comment (wwcchh0123) [reply internal/linters/go/gofmt/gofmt_test.go]: updated

**Pull request merged**